ansible_base/jwt_consumer/common/auth.py

```diff
@@ -265,27 +278,118 @@ def validate_token(self, token, decryption_key):
+            pass
+
+        # The resource was missing so we need to create its stub
+        if content_type == 'team':
+            # For a team we first have to make sure the org is there
+            org_id = data['org']
+            organization_data = self.token['objects'][self.org_content_type.model][org_id]
+
+            # Now that we have the org we can build a team
+            resource, org = self.get_or_create_resource(self.org_content_type.model, organization_data)
+            team, created = self.team_content_type.model_class().objects.get_or_create(
+                name=data['name'], organization=org, defaults={'description': 'Waiting for resource sync'}
+            )
+            team.save()
+            if created:
+                logger.warning(f"Created team {data['name']}")
+            team.resource.ansible_id = object_ansible_id
+            team.resource.save()
+            return team.resource, team
+        elif content_type == 'organization':
+            org, created = self.org_content_type.model_class().objects.get_or_create(name=data['name'], defaults={'description': 'Waiting for resource sync'})
+            org.save()
+            logger.warning(f"Created organization {data['name']}")
+            org.resource.ansible_id = object_ansible_id
+            org.resource.save()
+            return org.resource, org
+        else:
+            logger.error(f"build_resource_stub does not know how to build an object of type {type}")
+            return None, None
+
 
 class JWTAuthentication(BaseAuthentication):
     map_fields = default_mapped_user_fields
+    use_rbac_permissions = False
+
+    def __init__(self):
+        self.common_auth = JWTCommonAuth(self.map_fields)
 
     def authenticate(self, request):
-        common_auth = JWTCommonAuth(self.map_fields)
-        user, token = common_auth.parse_jwt_token(request)
+        logger.debug("Calling JWTAuthetnication.authenticate")
+        self.common_auth.parse_jwt_token(request)
 
-        if user:
-            self.process_user_data(user, token)
-            self.process_permissions(user, token.get("claims", None), token)
+        if self.common_auth.user:
+            self.process_user_data()
+            self.process_permissions()
 
-            return user, None
+            return self.common_auth.user, None
         else:
             return None
 
-    def process_user_data(self, user, token):
-        common_auth = JWTCommonAuth(self.map_fields)
-        common_auth.map_user_fields(user, token)
+    def process_user_data(self):
+        self.common_auth.map_user_fields()
 
-    def process_permissions(self, user, claims, token):
-        logger.info("process_permissions was not overridden for JWTAuthentication")
+    def process_permissions(self):
+        if self.use_rbac_permissions:
+            self.common_auth.process_rbac_permissions()
+        else:
+            logger.info("process_permissions was not overridden for JWTAuthentication")
```

Comment:
Maybe message could say that RBAC is switched off, doing nothing or something like that.

Reply:
The initial intent is that every service would have a custom class extending this one and implement their own process_permissions to work however it needed to. Since then we've implemented RBAC and add the method that works with standard RBAC. So the patten is that services will have to make a conscious decision to use the standard RBAC processor. So this is intended really more as a "he dev you didn't do this" whether RBAC is installed or not.